File: pr1-java/services/src/main/java/pr1Java/services/thrift/datatransfer/DtoUtils.java
package pr1Java.services.thrift.datatransfer;

import pr1Java.model.Game;
import pr1Java.model.User;
import pr1Java.services.Configuration;

public class DtoUtils {

    public static User toUser(UserDto userDto) {
        Configuration.logger.traceEntry("entering with {}", userDto);

        String username = userDto.getUsername();
        String password = userDto.getPassword();
        User user = new User(username, password);

        Configuration.logger.traceExit(user);
        return user;
    }

    public static UserDto toDto(User user) {
        Configuration.logger.traceEntry("entering with {}", user);

        String username = user.getUsername();
        String password = user.getPassword();
        UserDto userDto = new UserDto(username, password);

        Configuration.logger.traceExit(userDto);
        return userDto;
    }

    public static Game toGame(GameDto gameDto) {
        Configuration.logger.traceEntry("entering with {}", gameDto);

        Integer id = gameDto.getGameId();
        String name = gameDto.getName();
        String homeTeam = gameDto.getHomeTeam();
        String awayTeam = gameDto.getAwayTeam();
        Integer availableSeats = gameDto.getAvailableSeats();
        Integer seatCost = gameDto.getSeatCost();
        Game game = new Game(id, name, homeTeam, awayTeam, availableSeats, seatCost);

        Configuration.logger.traceExit(game);
        return game;
    }

    public static GameDto toDto(Game game) {
        Configuration.logger.traceEntry("entering with {}", game);

        Integer id = game.getId();
        String name = game.getName();
        String homeTeam = game.getHomeTeam();
        String awayTeam = game.getAwayTeam();
        Integer availableSeats = game.getAvailableSeats();
        Integer seatCost = game.getSeatCost();
        GameDto gameDto = new GameDto(id, name, homeTeam, awayTeam, availableSeats, seatCost);

        Configuration.logger.traceExit(gameDto);
        return gameDto;
    }
}

File: pr1-cs/Services/Thrift/DataTransfer/DtoUtils.cs
using Model;

namespace Services.Thrift.DataTransfer
{
    public class DtoUtils
    {
        public static User ToUser(UserDto userDto)
        {
            var username = userDto.Username;
            var password = userDto.Password;
            var user = new User(username, password);

            return user;
        }

        public static UserDto ToDto(User user)
        {
            var username = user.Username;
            var password = user.Password;
            var userDto = new UserDto {Username = username, Password = password};

            return userDto;
        }

        public static Game ToGame(GameDto gameDto)
        {
            var id = gameDto.GameId;
            var name = gameDto.Name;
            var homeTeam = gameDto.HomeTeam;
            var awayTeam = gameDto.AwayTeam;
            var availableSeats = gameDto.AvailableSeats;
            var seatCost = gameDto.SeatCost;
            var game = new Game(id, name, homeTeam, awayTeam, availableSeats, seatCost);

            return game;
        }

        public static GameDto ToDto(Game game)
        {
            var id = game.Id;
            var name = game.Name;
            var homeTeam = game.HomeTeam;
            var awayTeam = game.AwayTeam;
            var availableSeats = game.AvailableSeats;
            var seatCost = game.SeatCost;
            var gameDto = new GameDto
            {
                GameId = id, Name = name, HomeTeam = homeTeam, AwayTeam = awayTeam, AvailableSeats = availableSeats,
                SeatCost = seatCost
            };

            return gameDto;
        }
    }
}